### PR TITLE
Milestone 10: Add authorization

### PR DESCRIPTION
**Pull request by Debas-31**
# Rails Blog App: Add authorization
>In this milestone of the project we implemented the requirements:
- [x] Install CanCanCan in your project.
- [x] Add a role column to the users table. Remember to use a migration for this.
- [x] A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
   - For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
- [x] A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
   - For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,4 +1,6 @@
 class PostsController < ApplicationController
+  before_action :authenticate_user!, only: %i[create destroy]
+
```
Comment (julie-ify) [inline app/controllers/posts_controller.rb]: - [ ] Kindly use `CanCanCan` for Post authorization in the `Post Controller`

```ruby
  class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```

File: db/migrate/20220721103300_add_role_to_users.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddRoleToUsers < ActiveRecord::Migration[7.0]
+  def change
+    add_column :users, :role, :string, default: 'normal', null: false
+  end
+end
```
Comment (julie-ify) [inline db/migrate/20220721103300_add_role_to_users.rb:5]: - [ ] I received the attached error when I tried migrating your database with the command; `rails db:migrate` after creating the database. Kindly fix the error so we can test the `CanCanCan` functionality on your app.

Code_gkCH0U1QfL

File: app/models/ability.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    user ||= User.new # guest user (not logged in)
+    if user.admin?
+      can :manage, :all
+    else
+      can :manage, Post, author_id: user.id
+      can :manage, Comment, author_id: user.id
+      can :read, :all
+    end
+  end
+end
```
Comment (julie-ify) [inline app/models/ability.rb:14]: - [ ] Good job

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Debas-31,

Good job so far!

## Highlights!
- GitFlow ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (Debas-31) [reply app/controllers/posts_controller.rb]: Thank you for the feedback implemented the required change

File: db/migrate/20220721103300_add_role_to_users.rb (hunk 2 above)
Comment (Debas-31) [reply db/migrate/20220721103300_add_role_to_users.rb:5]: Thank you for the feedback. We fixed the error.

**Review by prolajumokeoni — APPROVED** [review]
Hi @Debas-31, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- No linter errors ✅ 
- Git flow was used correctly 🚀 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**